src/org/labkey/test/components/ui/navigation/apps/ServerNotificationItem.java

```diff
@@ -0,0 +1,218 @@
+        }
+        else
+        {
+            return "";
+        }
+    }
+
+    /**
+     * The displayed username while an import is in progress. Once the import completes this is not present.
+     *
+     * @return The username displayed, empty string if not present.
+     */
+    public String getUserName()
+    {
+        if(elementCache().userName.isDisplayed())
+        {
+            return elementCache().userName.getText();
+        }
+        else
+        {
+            return "";
+        }
+    }
+
+    /**
+     * The text for the 'View' link. Will not be present during an import but is visible for both a successful and unsuccessful import.
+     *
+     * @return Text of the 'View' link, empty string if not present.
+     */
+    public String getViewLinkText()
+    {
+        if(elementCache().link.isDisplayed())
+        {
+            return elementCache().link.getText();
+        }
+        else
+        {
+            return "";
+        }
+    }
+
+    /**
+     * <p>
+     *     Click the 'View' link in the message entry.
+     * </p>
+     * <p>
+     *     This may cause a navigation to occur. For example if the import failed this will take you to the error
+     *     report, if a success it will go someplace else depending upon the app.
+     * </p>
+     * <p>
+     *     This will wait until the url changes. It is up to the consuming code to know which page the navigation went to.
+     * </p>
+     */
+    public void clickViewLink()
+    {
+        String currentUrl = getDriver().getCurrentUrl().toLowerCase();
+        String targetUrl = elementCache().link.getAttribute("href").toLowerCase();
+
+        elementCache().link.click();
+
+        // Check if currently on the target page.
+        if(!currentUrl.contains(targetUrl))
+        {
+            WebDriverWrapper.waitFor(()->!getDriver().getCurrentUrl().toLowerCase().contains(currentUrl),
+                    "Clicking the 'View' link in the notification did not navigate in time.", 5_000);
+        }
+
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends WebDriverComponent.ElementCache
+    {
+        private final WebElement status = Locator.tagWithClass("i", "fa").findWhenNeeded(this);
+        private final WebElement message = Locator.tagWithClass("span", "server-notification-message").findWhenNeeded(this);
+        private final WebElement date = Locator.tagWithClass("div", "server-notification-data").findWhenNeeded(this);
+        private final WebElement userName = Locator.tagWithClass("span", "server-notification-data").findWhenNeeded(this);
+        private final WebElement link = Locator.tagWithClass("span", "server-notifications-link").childTag("a").findWhenNeeded(this);
+    }
+
+
+    public static class AppNotificationEntryFinder extends WebDriverComponentFinder<ServerNotificationItem, AppNotificationEntryFinder>
```

Comment:
With such a simple locator, I'd make this package protected so that only the `ServerNotificationMenu` can access it.
```suggestion
   static class AppNotificationEntryFinder extends WebDriverComponentFinder<ServerNotificationItem, AppNotificationEntryFinder>
```

Reply:
Changed.